src/UseCases/Food/CreateFoodUseCase.java

```diff
@@ -51,7 +51,7 @@ public CreateFoodUseCase(FoodRepository foodRepo, VendorRepository vendorRepo,
     public boolean createFood(String token, String shopId, String name, String desc,
                               float price, List<ISingleton> singletons){
 
-        IVendor vendor = vendorRepository.getVendorFromToken(token);
+        IVendor vendor = (IVendor)vendorRepository.getUserFromToken(token);
```

Comment:
same here

Reply:
Vendor repository extends user repository so we have the getUserFromToken method